skins/cat17/src/scripts/donationComments.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @licence GNU GPL v2+
+ * @author Gabriel Birke <gabriel.birke@wikimedia.de>
+ */
+
+/**
+ * Load and render current donation comments
+ *
+ * @constructor
+ */
+var DonationComments = function ( commentContainer, paginationContainer ) {
+	this.commentContainer = commentContainer;
+	this.paginationContainer = paginationContainer;
+};
+
+$.extend( DonationComments.prototype, {
+	init: function () {
+		this.commentContainer.html( 'Spendenkommentare werden geladen ...' );
+		this.paginationContainer.find( '.first' ).bind( 'click', $.proxy( this.goToFirstPage, this ) );
+		this.paginationContainer.find( '.last' ).bind( 'click', $.proxy( this.goToLastPage, this ) );
+		this.paginationContainer.find( '.prev' ).bind( 'click', $.proxy( this.goToPrevPage, this ) );
+		this.paginationContainer.find( '.next' ).bind( 'click', $.proxy( this.goToNextPage, this ) );
+		this.currentPage = 0;
+		this.itemsPerPage = 10;
+		this.numPages = 1;
+		this.update();
+	},
+
+	update: function () {
+		var self = this;
+		$.ajax( '../list-comments.json?n=100&anon=1', {
+			dataType: 'json',
+			success: function ( data ) {
+				self.numPages = Math.ceil( data.length / self.itemsPerPage );
+				self.commentContainer.html( self.renderHtml( data ) );
+				self.updatePagination();
+			}
+		} );
+	},
+
+	renderHtml: function ( data ) {
```

Comment:
Strictly speaking this does not render, but return mixed type results, either a string, or object containing HTML elements - `.html()` just happens to get along with them.